apps/publications/src/app/dialogs/add-authors-dialog/add-authors-dialog.component.ts

```diff
@@ -45,21 +31,33 @@ export class AddAuthorsDialogComponent implements OnInit {
   selection = new SelectionModel<Author>(false, []);
   reloadTable = false;
 
+  constructor(
+    private dialogRef: MatDialogRef<AddAuthorsDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) private data: AddAuthorsDialogData,
+    private notificator: NotificatorService,
+    private cabinetService: CabinetManagerService,
+    private translate: TranslateService
+  ) {
+    translate
+      .get('DIALOGS.ADD_AUTHORS.SUCCESS_MESSAGE')
+      .subscribe((value: string) => (this.successMessage = value));
+    this.publicationId = data.publicationId;
+    this.alreadyAddedAuthors = data.alreadyAddedAuthors;
+  }
+
   ngOnInit(): void {
     this.searchControl = new FormControl('', [
       Validators.required,
       Validators.pattern('.*[\\S]+.*'),
     ]);
   }
 
-  onSearchByString() {
+  onSearchByString(): void {
     this.searchLoading = true;
     const removeAuthors = [...this.alreadyAddedAuthors, ...this.authorsToAdd];
-    this.cabinetService.findNewAuthors(this.searchControl.value).subscribe(
+    this.cabinetService.findNewAuthors(this.searchControl.value as string).subscribe(
       (authors) => {
-        authors = authors.filter(
-          (item) => removeAuthors.map((author) => author.id).indexOf(item.id) <= -1
-        );
+        authors = authors.filter((item) => removeAuthors.includes(item));
```

Comment:
This filter is not working
when i put to searchfield something like "martin" the backend post request finds many possible author but the table says No authors were found.
![image](https://user-images.githubusercontent.com/32871167/165150303-3b4c312b-bb55-4c7b-9f2a-0753b8a48efd.png)

Reply:
fixed filtering